### PR TITLE
[PackageType] Add PackageType to catalog

### PR DESCRIPTION
**Pull request by ryuyu**
Adds packageType information in to catalog.

### CODE REVIEW

File: tests/NgTests/Data/TestCatalogEntries.resx [hunk 1]
```diff
@@ -850,6 +866,10 @@
       "@id": "packageEntry",
       "@container": "@set"
     },
+		"packageTypes": {
```
Comment (joelverhagen) [inline tests/NgTests/Data/TestCatalogEntries.resx]: Indentation is inconsistent here

File: src/Catalog/xslt/nuspec.xslt [hunk 2]
```diff
@@ -290,39 +334,39 @@
...
-                </ng:assembly>
-                <ng:targetFramework>
-                  <xsl:value-of select="." />
-                </ng:targetFramework>
-              </rdf:Description>
-            </ng:frameworkAssemblyGroup>
-          </xsl:for-each>
-        </xsl:when>
-        <xsl:otherwise>
+    <xsl:choose>
```
Comment (joelverhagen) [inline src/Catalog/xslt/nuspec.xslt]: nit: Can we revert these indentation changes? makes the diff noisy

File: tests/CatalogTests/PackageCatalogItemTests.cs [hunk 3]
```diff
@@ -27,6 +27,11 @@ public class PackageCatalogItemTests
         [InlineData("InvalidDependencyVersionRange.0.1.0")] //A dependency with a version range that is invalid
         [InlineData("MissingDependencyVersionRange.0.1.0")] // A dependency with no version range attribute
         [InlineData("WhitespaceDependencyVersionRange.0.1.0")] // A dependency with a version range that is whitespace
+        [InlineData("PackageTypeCollapseDuplicate")]
```
Comment (joelverhagen) [inline tests/CatalogTests/PackageCatalogItemTests.cs]: Same type different version?
Duplicate type name with different case?
Same type different version by System.Version but not SemVer (1.0, 1.0.0, 1.0.0.0)?
Whitespace around name? Whitepace around version?

It sucks we have to have the full nupkg and JSON data for each of these test variants. That's a lot of noise for small changes between tests.

File: tests/NgTests/Data/TestCatalogEntries.resx (hunk 1 above)
Comment (ryuyu) [reply tests/NgTests/Data/TestCatalogEntries.resx]: Corrected all tabs to spaces. Look consistent for me now. Hopefully will translate correctly here.

File: src/Catalog/xslt/nuspec.xslt (hunk 2 above)
Comment (ryuyu) [reply src/Catalog/xslt/nuspec.xslt]: I used VS auto format here. It's a bit noisy now, but it's at least internally consistent. Hopefully, we can just use autoformat for this doc any time we need to make changes going forward.

**Pull request merged**